ProcessMaker/Http/Resources/V1_1/TaskResource.php

```diff
@@ -0,0 +1,168 @@
+        'requestor' => ['id', 'first_name', 'last_name', 'email'],
+        'processRequest' => ['id', 'process_id', 'status'],
+        'draft' => ['id', 'task_id', 'data'],
+        'screen' => ['id', 'config'],
+        'process' => ['id', 'name'],
+    ];
+
+    /**
+     * Transform the resource into an array.
+     *
+     * @param  \Illuminate\Http\Request
+     * @return array
+     */
+    public function toArray($request)
+    {
+        $array = [
+            'id' => $this->id,
+            'element_name' => $this->element_name,
+            'element_id' => $this->element_id,
+            'due_at' => $this->due_at,
+            'advancedStatus' => $this->advanceStatus,
+        ];
+
+        return $this->processInclude($request, $array);
+    }
+
+    private static function addRelationshipKeyColumn(Builder $query, string $relationship): bool
+    {
+        $model = $query->getModel();
+        if (!method_exists($model, $relationship)) {
+            return false;
+        }
+        $relationshipObject = $model->$relationship();
+        if (!($relationshipObject instanceof Relation)) {
+            return false;
+        }
+
+        if ($relationshipObject instanceof BelongsTo) {
+            $relationshipKey = $relationshipObject->getForeignKeyName();
+            $query->addSelect($relationshipKey);
+        }
+
+        return true;
+    }
+
+    private static function addRelationship(ProcessRequestToken $model, string $relationship): bool
+    {
+        if (!method_exists($model, $relationship)) {
+            return false;
+        }
+        $relationshipObject = $model->$relationship();
+        if (!($relationshipObject instanceof Relation)) {
+            return false;
+        }
+
+        $relationshipColumns = self::$defaultFieldsFor[$relationship] ?? ['id'];
+        $model->$relationship = $relationshipObject->select($relationshipColumns)->get();
+
+        return true;
+    }
+
+    public static function preprocessInclude(Request $request, Builder $query): self
+    {
+        foreach (self::$defaultFields as $field) {
+            $query->addSelect($field);
+        }
+
+        $include = $request->query('include', []);
+        if ($include) {
+            $include = explode(',', $include);
+        }
+        $include[] = 'process';
+        $include[] = 'advanceStatus';
+
+        if (in_array('data', $include)) {
+            $query->addSelect('process_request_id');
+        }
+
+        foreach (self::$includeMethods as $key) {
+            if (!in_array($key, $include)) {
+                continue;
+            }
+
+            self::addRelationshipKeyColumn($query, $key);
+        }
+
+        $model = $query->first();
+        foreach (self::$includeMethods as $key) {
+            if (!in_array($key, $include)) {
+                continue;
+            }
+
+            self::addRelationship($model, $key);
+        }
+
+        return new static($query->first());
```

Comment:
maybe we can improve $query->first() was called two times


```

    $model = $query->first();
        foreach (self::$includeMethods as $key) {
            if (in_array($key, $include)) {
                self::addRelationship($model, $key);
            }
        }

        return new static($model);
```

Reply:
Very good catch!